spec/std/enumerable_spec.cr

```diff
@@ -262,7 +262,7 @@ describe "Enumerable" do
   describe "in_groups_of" do
     assert { [1, 2, 3].in_groups_of(1).should eq([[1], [2], [3]]) }
     assert { [1, 2, 3].in_groups_of(2).should eq([[1, 2], [3, nil]]) }
-    assert { ([] of Int32).in_groups_of(2).should eq([] of Array(Array(Int32 | Nil))) }
+    assert { ([] of Int32).in_groups_of(2).should eq([] of Array(Int32 | Nil)) }
```

Comment:
Not only is the test incorrect but the current in tree `==` is incorrect.  The current `==` identifies a 2 level deep empty array as equal to a 3 level deep empty array.  This PR fixes the broken behavior and corrects the test.

`[] of Array(Array(Int32 | Nil))` is a three level deep array.  `Array(Array(Array(Int32 | Nil)))`

Proof
```
a = [] of Array(Array(Int32 | Nil))
b = [] of Array(Int32 | Nil)
p a
p b
p a == b #=> says true but NOT true
```

`in_groups_of` works properly
```
# Both return Array(Array(Nil | Int32))
p ([1, 2, 3, 4] of Int32).in_groups_of(2) #=> [[1, 2], [3, 4]]
p ([] of Int32).in_groups_of(2)  #=> []
```